### ISSUE TITLE
Models Supported - Do you have a different model than this list? 

### ISSUE THREAD

**Issue by D-Jeffrey** [crowbarz/ha-pioneer_async#20]
These models I note either have custom configuration or people have noted that they are using that model and we can assume it works.  Are there any other models that people have used this on?

|  SC | SC  |SC   | SC-LX  | VSX | VSX | VSX |
|--- |--- |--- |---    |--- |--- | ---
|	SC-71	|	SC-1223	|	SC-2023	|	SC-LX77	|	VSX-45	| VSX-529		|	VSX-1025*
|	SC-75	|	SC-1228	|		|	SC-LX79	|		|	VSX-830	|	VSX-1028
|	SC-77	|	SC-1323	|		|	SC-LX87	|		|	VSX-923		|	VSX-1120
|		|	SC-1328	|		|		|		|	VSX-925	|	VSX-1123
|		|	SC-1523	|		|		|		|	VSX-930	|	VSX-1128
|		|	SC-1528	|		|		|		|		|

**Comment by D-Jeffrey** [comment]
Can you add a Label of 'Question'

**Comment by 11harveyj** [comment]
FYI, a lot of the configurations that were added in aiopioneer 0.2.0 were taken from the Pioneer docs, I can't guarantee that they will 100% work.

It would be worth asking the release year of each AVR and firmware version too if known. That way it helps when creating a configuration for that AVR as it can be cross referenced to the Pioneer docs.

**Comment by Toets** [comment]
Please add my old VSX-2021. Config flow didn't work for me but yaml does.

**Comment by 11harveyj** [comment]
@Toets can you provide your yaml config please?

**Comment by Toets** [comment]
media_player:
  - platform: pioneer_async
    name: AVR WK
    host: 10.100.15.22
    port: 8103

**Comment by 11harveyj** [comment]
Thats interesting, those options are configurable via the config flow. What happens when you configure with the config flow?

You'll need to change the port from the default value.

**Comment by D-Jeffrey** [comment]
@11harveyj I find that [hifiengine](https://www.hifiengine.com/manual_library/pioneer.shtml) is good for looking up older models. VSX-2021 is 2011/2012.  Since @crowbarz added the prompts in the flow for the port number in the Add interface, it can be adjusted (only at that time) without having to add configuration.yaml.  So far, I have found a couple of variances that make the device work differently - volume step, no standby enabled/supported and no response to ?SVB.

Reading the VSX-1021 manual, I see that you can adjust the port number.
`Home menu-> System Setup -> Network Setup -> `**`Port Number Settings`**
And in the manual, it suggests 23.  It also indicates that you can have up to 5 port numbers set.

@Toets is that the case on your receiver?  Can you add multiple port numbers? and it adjustable there?

Maybe there are some 'Adjust your Pioneer for best experience' notes that should be in the readme.

**Comment by D-Jeffrey** [comment]
@11harveyj I created a list of the possible models: 56 in all which includes the year it was made [Notes](https://github.com/D-Jeffrey/pioneer_vsx/blob/master/DeveloperNote.md) & [JSON](https://github.com/D-Jeffrey/pioneer_vsx/blob/master/pioneermodel.json)

**Comment by 11harveyj** [comment]
Looks good, thank you, just to note, my SC-LX57 supports network, network standby and pretty much all other commands too (it's what was used when I was building out all the commands for aiopioneer)

From my point of view all AVRs with a model number starting with SC-LX and newer than 2012 (maybe 2011) supports most features, I'm led to believe these were the "best of the best" 10~ yrs ago.

There's a good amount of docs for most AVRs starting SC too, the VSX AVRs are where things get tricky because not so much exists for those (such as yours for example).

I think it would be good to build a dev tool to send and capture as much protocol related information about an AVR as possible, mainly for these AVRs where we're not sure on whats supported and what isn't.

**Comment by D-Jeffrey** [comment]
I will have to see if there is a better repository out there that I can get a improved answer from.  Is there a way that we could push into to the 'analytics' to build up information on all the different models (properly) that choose to install this integration?  I wonder if there is a way, we could leverage this idea [processing](https://www.home-assistant.io/integrations/analytics/#data-storage--processing)?

I see in the data [[Analytics](https://analytics.home-assistant.io/)  I see there are: `390 | Pioneer | 221 (0.1%)` of the built in Pioneer model.  (I wonder how many people removed the default Pioneer and are trying other customs) like:
@mk3l @BIoob @lakee911 @CodingSpiderFox @leodirob @jtonk @tiagofreire-pt @user0x01 @realthk and so many more.)

**Comment by Toets** [comment]
@D-Jeffrey: Yes, ports can be added on my VSX-2021 as you described. Ever since I owned this AVR I experienced issues controlling over the network. I added a port long time ago while investigating the issue. Using telnet I can simply see the AVR gets unresponsive on a particular port after a while. A power cycle fixes the issue. It is an understatement to say the VSX-2021 is buggy...

In the config flow I change the host IP address and port number. Pressing Submit it immediately reports Cannot connect to AVR and input fields change back to default. Tired of troubleshooting due to buggy AVRI fell back to yaml and am able to somewhat make this work. ha-pioneer_async is a huge improvement over the standard pioneer integration but still far from stable in my scenario.

**Comment by 11harveyj** [comment]
@Toets there are some comments about the stability of the Pioneer LAN interface here https://github.com/crowbarz/aiopioneer/discussions/18

In essence, I think its due to the fact the AVR does not properly handle the termination of TCP network connections, with its limited memory they get to a point where they cannot handle any new connections as you have described. Do you use the Pioneer app at all or anything else connecting to the AVR? These issues seem to affect Pioneer AVRs regardless if you navigate to the web server on it, or use the serial connection.

**Comment by MahrWe** [comment]
I have the VSX-921, it's connected to my network via LAN. I cannot change the port, it's fixed to 8103. 
I can confirm the mentioned stability issues, this is why I moved from ha-pioneer/onkyo integration to this one. This has worked for me very well ever since.

**Comment by rarroyo6** [comment]
The VSX-823 can be added to the list. I have one running, and can verify that it works fine with this integration.
The sources have to be added manually, they are not automatically detected.

**Comment by DeckardCain2014** [comment]
Does anyone has the VSX-932 successfully setup? I can't connect with port 23 nor 8102

**Comment by crowbarz** [comment]
> Does anyone has the VSX-932 successfully setup? I can't connect with port 23 nor 8102

Try power cycling the AVR, especially if you have previously used the core pioneer integration.

Does the AVR respond to ping? Are you able to use the relevant Pioneer app to remotely control the AVR?

You can also try to manually connect to your AVR via `telnet` or `nc` to ensure that the AVR is responding on either port.

**Comment by jasongabler** [comment]
I have a VSX-LX503.  I can't even get the device to appear in the devices list after configuration and reboot. 

Here's my config:

```
media_player:
  - platform: pioneer_async
    name: Pioneer AVR
    host: 192.168.0.134
    port: 8102
    scan_interval: 60
    timeout: 5.0
    params:
      ignore_volume_check: true
```

Surely, part of the issue is that my receiver doesn't listen on that port:

```
% nmap 192.168.0.134 
Starting Nmap 7.94 ( https://nmap.org ) at 2023-08-12 17:39 PDT
Nmap scan report for 192.168.0.134
Host is up (0.0027s latency).
Not shown: 991 closed tcp ports (conn-refused)
PORT      STATE SERVICE
80/tcp    open  http
7000/tcp  open  afs3-fileserver
8008/tcp  open  http
8009/tcp  open  ajp13
8080/tcp  open  http-proxy
8443/tcp  open  https-alt
8888/tcp  open  sun-answerbook
9000/tcp  open  cslistener
10001/tcp open  scp-config
```

I tried all of those ports in the config. All of them gave something like or similar to:

```
2023-08-12 19:50:31.625 WARNING (MainThread) [homeassistant.components.media_player] Setup of media_player platform pioneer_async is taking over 10 seconds.
2023-08-12 19:50:41.633 ERROR (MainThread) [custom_components.pioneer_async.media_player] Could not open AVR connection: RuntimeError: Main Zone not found on AVR
2023-08-12 19:50:41.635 WARNING (MainThread) [homeassistant.components.media_player] Platform pioneer_async not ready yet: None; Retrying in background in 30 seconds
```

Any ideas? 

Thanks.

**Comment by pantherale0** [comment]
> I have a VSX-LX503. I can't even get the device to appear in the devices list after configuration and reboot.
> 
> Here's my config:
> 
> ```
> media_player:
>   - platform: pioneer_async
>     name: Pioneer AVR
>     host: 192.168.0.134
>     port: 8102
>     scan_interval: 60
>     timeout: 5.0
>     params:
>       ignore_volume_check: true
> ```
> 
> Surely, part of the issue is that my receiver doesn't listen on that port:
> 
> ```
> % nmap 192.168.0.134 
> Starting Nmap 7.94 ( https://nmap.org ) at 2023-08-12 17:39 PDT
> Nmap scan report for 192.168.0.134
> Host is up (0.0027s latency).
> Not shown: 991 closed tcp ports (conn-refused)
> PORT      STATE SERVICE
> 80/tcp    open  http
> 7000/tcp  open  afs3-fileserver
> 8008/tcp  open  http
> 8009/tcp  open  ajp13
> 8080/tcp  open  http-proxy
> 8443/tcp  open  https-alt
> 8888/tcp  open  sun-answerbook
> 9000/tcp  open  cslistener
> 10001/tcp open  scp-config
> ```
> 
> I tried all of those ports in the config. All of them gave something like or similar to:
> 
> ```
> 2023-08-12 19:50:31.625 WARNING (MainThread) [homeassistant.components.media_player] Setup of media_player platform pioneer_async is taking over 10 seconds.
> 2023-08-12 19:50:41.633 ERROR (MainThread) [custom_components.pioneer_async.media_player] Could not open AVR connection: RuntimeError: Main Zone not found on AVR
> 2023-08-12 19:50:41.635 WARNING (MainThread) [homeassistant.components.media_player] Platform pioneer_async not ready yet: None; Retrying in background in 30 seconds
> ```
> 
> Any ideas?
> 
> Thanks.

This is something for a separate issue, but you need to make sure you can use the iControlAV 5 app with your AVR model. It might be too new for this integration as it looks like it doesn't expose a serial connection via the LAN.

Make sure network standby is turned on and perhaps l auto standby is turned off.

If it's still not working, test with the onkyo integration and see if that works instead. Otherwise I'm all out of ideas.

**Comment by jasongabler** [comment]
Yeah, I established a long time ago that it doesn't work with the Onkyo
integration.  I know at some point that Onkyo purchased Pioneer, but it
took a while for Pioneer branded AVRs to start using the (far superior)
Onkyo protocols. I think the LX503 was one of the last to use the old
Pioneer communication.   FWIW, my receiver can be controlled via
https://github.com/miracle2k/onkyo-eiscp.

**Comment by D-Jeffrey** [comment]
@pantherale0, A couple things to note and try. 

`nmap` does not automatically scan all the ports, just well-known ports without parameters.

```
$ nmap 192.168.4.201
Starting Nmap 7.80 ( https://nmap.org ) at 2023-08-13 21:29 MDT
Nmap scan report for 192.168.4.201
Host is up (0.049s latency).
Not shown: 999 closed ports
PORT   STATE SERVICE
80/tcp open  http

Nmap done: 1 IP address (1 host up) scanned in 0.73 seconds
````

if you use the port number it will show 8102 as active.  
````
$ nmap 192.168.4.201 -p 1-9000
Starting Nmap 7.80 ( https://nmap.org ) at 2023-08-13 21:33 MDT
Nmap scan report for 192.168.4.201
Host is up (0.049s latency).
Not shown: 8998 closed ports
PORT     STATE SERVICE
80/tcp   open  http
8102/tcp open  kz-migr

Nmap done: 1 IP address (1 host up) scanned in 8.95 seconds
````

You can try this url and see if it gives a response. 
[http://192.168.0.134/StatusHandler.asp](http://192.168.0.134/StatusHandler.asp)  you can see some more of the different links you can use [My Pioneer Info](https://github.com/D-Jeffrey/Pioneer-info)

I would also check page 149 of [VSX-LX503 Manual](https://intl.pioneer-audiovisual.com/manuals/docs/28d8197e3064e03602937fe0afb9de235f4b60ab.pdf) and see if the port has been changed from the default 8102.  If you are using Control4, then turn off the app while you try and configure.  Some versions can not handle multiple apps connecting to it at the sametime.

I would take the configuration out of your `configuration.yaml` and just using the configuration option after you add the Pioneer AVR module to HA and add a device in the interface instead.  

Update on updated on what you find.

**Comment by BIoob** [comment]
I'm using it with a VSX-924

**Comment by jasongabler** [comment]
> @pantherale0, A couple things to note and try.
> 
> `nmap` does not automatically scan all the ports, just well-known ports without parameters.
> 
> ```

Thanks. I took your suggestion and the resulting list was longer. Unfortunately, the same result with none of the ports satisfying this HACS plugin:

```
% nmap 192.168.0.134 -p 1-11000
Starting Nmap 7.94 ( https://nmap.org ) at 2023-08-20 11:48 PDT
Nmap scan report for 192.168.0.134
Host is up (0.0031s latency).
Not shown: 10983 closed tcp ports (conn-refused)
PORT      STATE SERVICE
80/tcp    open  http
4545/tcp  open  worldscores
7000/tcp  open  afs3-fileserver
8008/tcp  open  http
8009/tcp  open  ajp13
8080/tcp  open  http-proxy
8443/tcp  open  https-alt
8554/tcp  open  rtsp-alt
8888/tcp  open  sun-answerbook
9000/tcp  open  cslistener
9013/tcp  open  unknown
10001/tcp open  scp-config
10109/tcp open  unknown
10110/tcp open  nmea-0183
10111/tcp open  nmea-onenet
10113/tcp open  netiq-endpoint
10120/tcp open  unknown
```

> You can try this url and see if it gives a response. http://192.168.0.134/StatusHandler.asp you can see some more of the different links you can use [My Pioneer Info](https://github.com/D-Jeffrey/Pioneer-info)

This returned a 404. 

> I would also check page 149 of [VSX-LX503 Manual](https://intl.pioneer-audiovisual.com/manuals/docs/28d8197e3064e03602937fe0afb9de235f4b60ab.pdf) and see if the port has been changed from the default 8102. If you are using Control4, then turn off the app while you try and configure. Some versions can not handle multiple apps connecting to it at the sametime.

What exactly did you see on pg 149?  I've scoured this entire manual many times and there is absolutely no mention of network ports, numerically or otherwise -- albeit this page does say "using an Internet browser", which alludes to HTTP (80). And, HTTP does work for accessing the minimal network config UI for which you can see only one view in that manual page. This UI is nothing more than the most basic network setup reporting and TCP/IP config, as well as minimal C4 and Airplay config (neither of which I use.)

> 
> I would take the configuration out of your `configuration.yaml` and just using the configuration option after you add the Pioneer AVR module to HA and add a device in the interface instead.

Hmmm, I've never seen this "configuration option".  My HA is completely up to date: OS, HA itself, all add-ons, plugins and HACS plugins, all at the latest. Here's the plugin's tile in the HACS main screen:

<img width="502" alt="image" src="https://github.com/crowbarz/ha-pioneer_async/assets/1828132/407e22f8-286b-42d1-bcdf-6c733cdf262f">

If I click on the tile I see:

<img width="902" alt="image" src="https://github.com/crowbarz/ha-pioneer_async/assets/1828132/1cf44e6a-ffe8-430b-ae9f-a85f4baef283">


> 
> Update on updated on what you find.

Unfortunately, nothing helpful so far.

**Comment by crowbarz** [comment]
@jasongabler you mentioned earlier that your VSX-LX503 _is_ controllable using https://github.com/miracle2k/onkyo-eiscp ? Unfortunately this integration doesn't implement the protocol used by that repo - from a quick scan of [the commands](https://github.com/miracle2k/onkyo-eiscp/blob/master/eiscp-commands.yaml) it doesn't look much like the Pioneer protocol. Also, the HA integration complaining about the main zone not found would suggest that it can connect to the AVR but isn't responding as expected to the Pioneer commands that the integration is sending.

The VSX-LX503 manual does mention (on page 131) that it may be controllable via the iControlAV5 application from Pioneer, though that could just be a copy/paste artefact. As @pantherale0 mentioned, if it works with that app then it should work with this integration.

I see there is an RS-232 port on the back of a VSX-LX503, so a (very) long shot would be to connect something that exposes the serial port as a network connection (such as a terminal server) and point the integration at that device. You'd want to confirm which protocol is being used by the AVR before going down that route, however.

**Comment by pantherale0** [comment]
@jasongabler that AVR looks to be too new if I'm reading that port scan right.

The eISCP port is 60128, which yours appears to be listening on and it also looks like some of the Chromecast ports are listed there too.

FWIW - it's probably safe to assume that AVRs with eISCP / Chromecast built-in this integration won't work. Unless the serial port exposes the old pioneer protocol, but I wouldn't go out of my way to buy hardware to test as it probably exposes onkyo's serial protocol which is quite different.

https://github.com/zarpli/onkyo

**Comment by jasongabler** [comment]
@crowbarz  and @pantherale0  ,

Thanks for the replies and insights. I thought my receiver was speaking a protocol that was too old. Seems like I had that backwards.  I'm pretty familiar with the eISCP stuff, at least enough to have made a utility to normalize volumes between zones (python running on UN*X).  I suppose I could finally make a foray into hacking HA and contributing to this project... or wait for the rest of you to upgrade your receivers :D   For now I'll have to stick with my Harmony Hub as the intermediary.

Thanks.

**Comment by lhassell** [comment]
I can confirm working on VSX-1021.

**Comment by cyberdiamond** [comment]
SC-LX73 confirmed working, thanks.
